app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,24 @@
+class PostsController < ApplicationController
+  def index
+    @posts = Post.where(author_id: params[:user_id])
+    @user = User.find(params[:user_id])
+  end
+
+  def show
+    @post = Post.find(params[:id])
+    @user = User.find(params[:user_id])
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.new(allowed_params)
```

Comment:
- [Optional] Instead of getting the  id of the current user, you can equally do something like:
`@post = @current_user.posts.new(allowed_params)`

Reply:
this is not working.
I used the old code because is working

Reply:
Hi @rachidelaid. In order to make the `current_user` work, you need to declare the method itself, or, if you use a gem like devise (which you will learn later about) you can do it automatically. What you can do if you want to implement what the previous reviewer mentioned, is going to the application controller, and declare a new method inside it, called `current_user`, and this method return `User.first` as you are going to have only one use right now. But as I mentioned, you will learn more ways to implement this later on, so this is not mandatory. Keep it up!